src/vttLib/transfer.py

```diff
@@ -0,0 +1,72 @@
+import io
+import logging
+import os
+
+import fontTools
+import fontTools.ttLib
+
+import vttLib
+
+logger = logging.Logger(__name__)
+
+
+def dump_to_file(font: fontTools.ttLib.TTFont, path: os.PathLike):
+    TABLES_TO_DUMP = ("TSI1", "TSI3", "TSI5", "maxp")
+    for table_tag in TABLES_TO_DUMP:
+        if table_tag not in font:
+            raise vttLib.VTTLibArgumentError(
+                "Table '%s' not found in input font" % table_tag
+            )
+
+    vttLib.normalize_vtt_programs(font)
+    font.saveXML(path, tables=TABLES_TO_DUMP)
+
+
+def merge_from_file(font: fontTools.ttLib.TTFont, path: os.PathLike):
+    if "TSI0" not in font:
+        font["TSI0"] = fontTools.ttLib.newTable("TSI0")
+    if "TSI2" not in font:
+        font["TSI2"] = fontTools.ttLib.newTable("TSI2")
+    font.importXML(path)
+
+
+def copy_from_ufo_data_to_file(ufo, path: os.PathLike):
+    font = fontTools.ttLib.TTFont()
+
+    font["maxp"] = maxp = fontTools.ttLib.newTable("maxp")
+    maxp.tableVersion = 0x00010000
+    maxp.maxComponentDepth = 0
+    maxp.maxComponentElements = max(len(g.components) for g in ufo)
+    maxp.maxCompositeContours = 0
+    maxp.maxCompositePoints = 0
+    maxp.maxContours = 0
+    maxp.maxFunctionDefs = 0
+    maxp.maxInstructionDefs = 0
+    maxp.maxPoints = 0
+    maxp.maxSizeOfInstructions = 0
+    maxp.maxStackElements = 0
+    maxp.maxStorage = 0
+    maxp.maxTwilightPoints = 0
+    maxp.maxZones = 1
+    maxp.numGlyphs = len(ufo)
+    if ".notdef" not in ufo:
+        maxp.numGlyphs += 1
```

Comment:
Why +1 if .notdef not in UFO?

Reply:
Because if increases numGlyphs by 1 when roundtripping (i.e. dumping the data again).